lightyear/src/transport/webtransport/client_wasm.rs

```diff
@@ -0,0 +1,176 @@
+#![cfg(target_family = "wasm")]
+//! WebTransport client implementation.
+use super::MTU;
+use crate::transport::{PacketReceiver, PacketSender, Transport};
+use anyhow::Context;
+use bevy::tasks::{IoTaskPool, TaskPool};
+use std::net::SocketAddr;
+use tokio::select;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TryRecvError;
+use tracing::{debug, error, info, trace};
+use web_sys::js_sys::{Array, Uint8Array};
+use web_sys::wasm_bindgen::JsValue;
+use web_sys::WebTransportHash;
+
+use base64::prelude::{Engine as _, BASE64_STANDARD};
+use xwt_core::prelude::*;
+use xwt_web_sys::{Connection, Endpoint};
+
+/// WebTransport client socket
+pub struct WebTransportClientSocket {
+    client_addr: SocketAddr,
+    server_addr: SocketAddr,
+    certificate_digest: String,
+}
+
+impl WebTransportClientSocket {
+    pub fn new(
+        client_addr: SocketAddr,
+        server_addr: SocketAddr,
+        certificate_digest: String,
+    ) -> Self {
+        Self {
+            client_addr,
+            server_addr,
+            certificate_digest,
+        }
+    }
+}
+
+fn js_array(values: &[&str]) -> JsValue {
+    return JsValue::from(
+        values
+            .into_iter()
+            .map(|x| JsValue::from_str(x))
+            .collect::<Array>(),
+    );
+}
+
+impl Transport for WebTransportClientSocket {
+    fn local_addr(&self) -> SocketAddr {
+        self.client_addr
+    }
+
+    fn listen(self) -> (Box<dyn PacketSender>, Box<dyn PacketReceiver>) {
```

Comment:
It is a bit confusing: why does `listen` connects to somewhere? Listening won't work in the browser obviously, so why is it implemented even?

Reply:
It could be more clear I think; but `listen` is the function I use to create the listener/receiver io channels.
I'll think about a better name